env0/resource_aws_credentials.go

```diff
@@ -72,6 +96,14 @@ func resourceAwsCredentialsCreate(ctx context.Context, d *schema.ResourceData, m
 
 	apiClient := meta.(client.ApiClientInterface)
 
+	valid, err := IsExternalIdValid(d, apiClient)
+	if err != nil {
+		return diag.Errorf("failed to validate external_id: %v", err)
+	}
+	if !valid {
+		return diag.Errorf("external_id is invalid")
```

Comment:
maybe add some info like - 'need to match pattern - ^[A-Za-z0-9+=\-.,@:/]{2,1000}$ '